kinto/core/authorization.py

```diff
@@ -86,6 +86,9 @@ def permits(self, context, principals, permission):
                 principals,
                 get_bound_permissions=self.get_bound_permissions)
             allowed = len(shared_records) > 0
+            if not allowed and context.force_empty_list:
+                allowed = True
+                context.forced_empty_list = True
```

Comment:
We need to detect that it was allowed while there were no shared_records.
If we don't know that we will endup having all the records as if we had the read permissions on buckets.